api/src/controllers/search.js

```diff
@@ -278,6 +278,18 @@ let searchCollection = async function(
     searchProperties = { $text: { $search: keywords, $caseSensitive: caseSensitive } };
   }
 
+  // has collection filter
+  // For this filter to work, we need to create a field and re-create the hasCollection match
+  // if we add to the initial match criteria though, it will break the initial match. So we
+  // flag it for addition later and remove the check from the "or" variable
+  let hasCollectionTest = false;
+  let hasCollection = null;
+  if (or && Object.prototype.hasOwnProperty.call(or, 'hasCollection')) {
+    hasCollectionTest = true;
```

Comment:
Remove?

Reply:
No, this is in to do a pre-check to see if we have a hasCollection filter. If we do, we need to remove it from the "or", otherwise the query to the DB will never return any results. We don't actually have a column for this filter so it has to be added dynamically.

we wouldn't have to do this step, but If we add the field before the match pipeline, the query will fail when you include text searches, because MongoDB aggregates require any $text pipeline to be the very first thing that runs. (Try a text search in test now, and you'll notice it fails)